deployment/src/main/java/org/wso2/testgrid/deployment/deployers/AWSDeployer.java

```diff
@@ -47,17 +48,22 @@ public String getDeployerName() {
     @Override
     public Deployment deploy(Deployment deployment) throws TestGridDeployerException {
         //wait for server startup
-        logger.info("Deploying the pattern..");
-        DeploymentValidator validator = new DeploymentValidator();
-        logger.info("Waiting for server startup..");
-        for (Host host : deployment.getHosts()) {
-            try {
-                new URL(host.getIp());
-            } catch (MalformedURLException e) {
-                continue;
+        try {
+            logger.info("Deploying the pattern..");
+            DeploymentValidator validator = new DeploymentValidator();
+            logger.info("Waiting for server startup..");
+            for (Host host : deployment.getHosts()) {
+                try {
+                    new URL(host.getIp());
+                } catch (MalformedURLException e) {
+                    continue;
```

Comment:
If the URL is malformed, 'continue' will cause a busy-loop. Shouldn't we throw the exception and stop the entire process?

Reply:
In this code, the correct output URLs to wait upon are identified by this method.
all other values are skipped by the MalformedURLException

Reply:
Got it.. Shall we log that as an error then?

Reply:
Done